### PR TITLE
Week 04: Ruby Methods, Classes, And Blocks

### PR DESCRIPTION
**Pull request by akabishau**
In this pull request:
- Completed the tasks based on lesson assignment

### CODE REVIEW

File: week-04/solutions/assertions.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+def assert(expected, result)
```
Comment (josiahandrew12) [inline week-04/solutions/assertions.rb:1]: 👍

File: week-04/solutions/block_function.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+def do_calc
```
Comment (josiahandrew12) [inline week-04/solutions/block_function.rb:1]: 👍

File: week-04/solutions/divisible.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+range_input = (1..100)
```
Comment (josiahandrew12) [inline week-04/solutions/divisible.rb:1]: Nice job with giving different solutions but next time just give a single solution that you find the most efficient.

File: week-04/solutions/hangman.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+require_relative "assertions"
+
+def hangman(word, letters)
```
Comment (josiahandrew12) [inline week-04/solutions/hangman.rb:1]: Is this function is not being called and only used it in test? We should calling the function with inputs as stated in the instructions.

File: week-04/solutions/sums.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+class Sum1
```
Comment (josiahandrew12) [inline week-04/solutions/sums.rb:1]: 👍

File: week-04/solutions/hash_to_array.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+def ask_question
```
Comment (josiahandrew12) [inline week-04/solutions/hash_to_array.rb:1]: 👍

File: week-04/solutions/hangman.rb (hunk 4 above)
Comment (akabishau) [reply week-04/solutions/hangman.rb:1]: Updated

**Comment by akabishau** [comment]
hey @josiahandrew12, I don't think I saw the message this assignment was successful (initial status - Review Needed). I fixed the part that I think required revision. Could you please confirm that this PR is complete and I can merge it? Thank you!

**Review by josiahandrew12 — APPROVED** [review]

**Comment by josiahandrew12** [comment]
> hey @josiahandrew12, I don't think I saw the message this assignment was successful (initial status - Review Needed). I fixed the part that I think required revision. Could you please confirm that this PR is complete and I can merge it? Thank you!

Just approved, will get this requested to marked as successfully today.

**Pull request merged**